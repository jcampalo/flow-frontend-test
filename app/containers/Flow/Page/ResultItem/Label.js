import styled from 'styled-components';

export default styled.label`
  font-size: 1em;
  color: ${({ error }) => error ? '#E91E63' : '#07DCA5'};
`;
