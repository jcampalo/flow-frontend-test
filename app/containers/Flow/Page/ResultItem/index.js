import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from '../../messages';
import Label from './Label';
import Wrapper from './Wrapper';

const ResultItem = props => {
  const { result, title } = props;

  if (typeof result === 'boolean') {
    return (
      <Wrapper>
        <FormattedMessage {...messages[result ? 'pass' : 'fail']} values={{ title }}>{
          text => (
            <Label error={!result}>{text}</Label>
          )
        }
        </FormattedMessage>
      </Wrapper>
    );
  }

  return (
    <Wrapper>
      <Label>{title}</Label>
    </Wrapper>
  );
};

ResultItem.propTypes = {
  result: PropTypes.bool,
  title: PropTypes.string
};

export default ResultItem;
