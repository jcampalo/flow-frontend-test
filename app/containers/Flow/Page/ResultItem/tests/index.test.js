import React from 'react';
import { create } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import ResultItem from '../index';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <IntlProvider {...props}>
      <ResultItem {...finalProps} />
    </IntlProvider>
  );
};

describe('<ResultItem />', () => {
  it('should render component ', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when result pass', () => {
    const props = {
      result: true,
      title: 'title'
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when result fail', () => {
    const props = {
      result: false,
      title: 'title'
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when ends', () => {
    const props = {
      title: 'title'
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
