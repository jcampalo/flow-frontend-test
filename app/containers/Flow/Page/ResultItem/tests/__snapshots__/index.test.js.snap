// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> should render component  1`] = `
<div
  className="Wrapper-dyuznp iDoJpU"
>
  <label
    className="Label-ghyrbC gTecle"
  />
</div>
`;

exports[`<ResultItem /> should render component when ends 1`] = `
<div
  className="Wrapper-dyuznp iDoJpU"
>
  <label
    className="Label-ghyrbC gTecle"
  >
    title
  </label>
</div>
`;

exports[`<ResultItem /> should render component when result fail 1`] = `
<div
  className="Wrapper-dyuznp iDoJpU"
>
  <label
    className="Label-ghyrbC dLBDCo"
  >
    Rule title fail
  </label>
</div>
`;

exports[`<ResultItem /> should render component when result pass 1`] = `
<div
  className="Wrapper-dyuznp iDoJpU"
>
  <label
    className="Label-ghyrbC gTecle"
  >
    Rule title pass
  </label>
</div>
`;
